lib/stackConfiguration/src/main/java/org/sagebionetworks/gcp/SynapseSystemPropertiesGCPCredentialsProvider.java

```diff
@@ -0,0 +1,26 @@
+package org.sagebionetworks.gcp;
+
+import java.util.Properties;
+
+import org.sagebionetworks.PropertyProvider;
+
+/**
+ * An AWS credential provider that will look for:
+ * 'org.sagebionetworks.stack.iam.id' and 'org.sagebionetworks.stack.iam.key' on
+ * the Java System.properties.
+ *
+ */
+public class SynapseSystemPropertiesGCPCredentialsProvider extends AbstractSynapseGcpCredentialsProvider {
+
+	Properties properties;
```

Comment:
make abstract class have a final Properties field and set it in the abstract class's constructors. 
and call super(properties)